include/lsst/meas/astrom/ScaledPolynomialTransformFitter.h

```diff
@@ -0,0 +1,275 @@
+     *                         rejected as outliers.
+     */
+    static ScaledPolynomialTransformFitter fromMatches(
+        int maxOrder,
+        afw::table::ReferenceMatchVector const & matches,
+        afw::image::Wcs const & initialWcs,
+        double intrinsicScatter
+    );
+
+    /**
+     *  Initialize a fit that inverts an existing transform by evaluating and
+     *  fitting to points on a grid.
+     *
+     *  @param[in] maxOrder         Maximum polynomial order for the fit.
+     *
+     *  @param[in] bbox             Bounding box for the grid in the input
+     *                              coordinates of the toInvert transform
+     *                              (or equivalently the output coordinates
+     *                              of the transform to be fit).
+     *
+     *  @param[in] nGridX           Number of grid points in the X direction.
+     *
+     *  @param[in] nGridY           Number of grid points in the Y direction.
+     *
+     *  @param[in] toInvert         Transform to invert
+     *
+     *  This initializes the data catalog with the following fields:
+     *   - output:  grid positions passed as input to the toInvert transform,
+     *              treated as output data points when fitting its inverse.
+     *   - input:   grid positions generated as the output of the toInvert
+     *              transform, treated as input data points when fitting its
+     *              inverse.
+     *   - model:   best-fit-transformed input points.
+     *
+     *  The updateIntrinsicScatter and rejectOutliers methods cannot be used
+     *  on fitters initialized using this method.
+     */
+    static ScaledPolynomialTransformFitter fromGrid(
+        int maxOrder,
+        afw::geom::Box2D const & bbox,
+        int nGridX, int nGridY,
+        ScaledPolynomialTransform const & toInvert
+    );
+
+    /**
+     *  Perform a linear least-squares fit of the polynomial coefficients.
+     *
+     *  @param[in]  order    The maximum order of the polynomial transform.
+     *                       If negative (the default) the maxOrder from
+     *                       construction is used.
+     */
+    void fit(int order=-1);
+
+    /**
+     *  Update the 'model' field in the data catalog using the current
+     *  best-fit transform.
+     */
+    void updateModel();
+
+    /**
+     *  Infer the intrinsic scatter in the offset between the data points
+     *  and the best-fit model.
+     *
+     *  The scatter is calculated as the RMS scatter that maximizes the
+     *  likelihood of the current positional offsets (assumed fixed)
+     *  assuming the per-data-point uncertainties are the quadrature sum
+     *  of the intrinsic scatter and that source's centroid measurement
+     *  uncertainty.
+     */
+    double updateIntrinsicScatter();
+
+    /**
+     *  Return the current estimate of the intrinsic scatter.
+     */
+    double getIntrinsicScatter() const { return _intrinsicScatter; }
```

Comment:
The intrinsic scatter is actually included in the uncertainty on the data points, so one might want to call this before `updateIntrinsicScatter` to obtain the scatter used in the last call to `fit` or `rejectOutliers`.  Calling it after `updateIntrinsicScatter` tells you the scatter that will be used in the next call to `fit` or `rejectOutliers`.  (I've added that to the docs for `getIntrinsicScatter`).